django/contrib/postgres/aggregates/statistics.py

```diff
@@ -0,0 +1,120 @@
+from django.db.models import F, FloatField, IntegerField
+from django.db.models.aggregates import Aggregate
+from django.db.models.expressions import Value
+from django.utils import six
+
+__all_ = [
+    'CovarPop', 'Corr', 'RegrAvgX', 'RegrAvgY', 'RegrCount', 'RegrIntercept',
+    'RegrR2', 'RegrSlope', 'RegrSXX', 'RegrSXY', 'RegrSYY',
+]
+
+
+class StatFunc(Aggregate):
+    template = "%(function)s(%(y)s, %(x)s)"
+    _num_expression_alias = 'num'
+
+    def __init__(self, y, x, output_field=FloatField()):
+        if not x or not y:
+            raise TypeError('Both X and Y must be provided.')
+        if (not isinstance(x, (six.text_type, six.string_types)) or
+                not isinstance(y, (six.text_type, six.string_types))):
+            raise ValueError('X and Y must be a string.')
```

Comment:
Before it will go to database, we need to decide what expression it will be (`F()` or `Value()`).
Because if we will pass an integer to `F()`, it will raise an exception that it is not a field name.
But we need to allow integer to be passed to final SQL-query, because postgres allow integer in statistic functions.